<launch>
  <!-- Get environment variables -->
  <arg name="venv" default="/usr/bin/python" doc="Path to python environment."/>
  <env name="LD_PRELOAD" value="/usr/lib/x86_64-linux-gnu/libffi.so.7" />

  <!-- Get arguments -->
  <arg name="model_path" default="$(find grounding_sam_ros)/weights/groundingdino_swint_ogc.pth" doc="Detection Model Path [default: weights/groundingdino_swint_ogc.pth]" />
  <arg name="config" default="$(find grounding_sam_ros)/config/GroundingDINO_SwinT_OGC.py" doc="Config path [default: src/groundingdino/config/GroundingDINO_SwinT_OGC.py]" />
  <arg name="sam_model" default="SAM-L" doc="Can be either SAM-B, SAM-L or SAM-H."/>
  <arg name="box_threshold" default="0.3" doc="Min confidence threshold for box." />
  <arg name="text_threshold" default="0.25" doc="Min confidence threshold for text." />
  
  <!-- Get SAM checkpoint -->
  <arg name="sam_checkpoint" value="$(find grounding_sam_ros)/weights/sam_vit_h_4b8939.pth" if="$(eval sam_model == 'SAM-H')" />
  <arg name="sam_checkpoint" value="$(find grounding_sam_ros)/weights/sam_vit_l_0b3195.pth" if="$(eval sam_model == 'SAM-L')" />
  <arg name="sam_checkpoint" value="$(find grounding_sam_ros)/weights/sam_vit_b_01ec64.pth" if="$(eval sam_model == 'SAM-B')" />

  <!-- Run Contact-GraspNet -->
  <node name="grounding_sam_ros" pkg="grounding_sam_ros" type="gsam.py" launch-prefix = "$(arg venv)" output="screen" >
      <param name="model_path" type="str" value="$(arg model_path)" />
      <param name="config" type="str" value="$(arg config)" />
      <param name="sam_checkpoint" type="str" value="$(arg sam_checkpoint)" />
      <param name="sam_model" type="str" value="$(arg sam_model)" />
      <param name="box_threshold" type="double" value="$(arg box_threshold)" />
      <param name="text_threshold" type="double" value="$(arg text_threshold)" />
  </node>
</launch>